src/main/groovy/org/cristalise/dev/test/scenario/ShiroPermissionTestIT.groovy

```diff
@@ -0,0 +1,43 @@
+package org.cristalise.dev.test.scenario
+
+import org.cristalise.kernel.test.KernelScenarioTestBase
+import org.junit.Test
+
+
+/**
+ *
+ *
+ */
+class ShiroPermissionTestIT extends KernelScenarioTestBase {
+    
+    @Test
+    public void 'Job is only given to the Agent with the proper Permission'() {
+        def oper1 = Agent("oper1-$timeStamp") {
+            Roles {
+                Role(name: 'oper') {
+                    Permission('test:EA1')
+                }
+            }
+        }
+
+        def clerk1 = Agent("clerk1-$timeStamp") {
+            Roles {
+                Role(name: 'clerk') {
+                    Permission('test:EA2')
+                }
+            }
+        }
```

Comment:
- Test defining a role in one Agent closure and using it in another Agent
- Test a negative case: execution of a job is denied because of insufficient permissions